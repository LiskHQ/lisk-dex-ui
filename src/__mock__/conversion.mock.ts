export const mockConversionRates = {
  LSK: {
    USD: 0.8863,
    DEX: 0.1,
  },
  DEX: {
    USD: 0.0886
  }
};