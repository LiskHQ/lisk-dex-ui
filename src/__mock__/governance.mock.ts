import { ProposalType } from "consts";
import { IPoolItem, IProposal } from "models";

export const mockProposal: IProposal = {
  author: "Author",
  proposalType: ProposalType.PoolIncentivization,
  poolID: "1",
  multiplier: 8,
  title: "title",
  summary: "summary",
  description: "description",
  link: "description"
}

export const mockPoolItems: IPoolItem[] = [
  {
    id: '1',
    chain1: 'LSK',
    chain2: 'DEU',
    rate: 0.3,
    amount: 12.2,
  },
  {
    id: '2',
    chain1: 'KLP',
    chain2: 'RGB',
    rate: 0.1,
    amount: 11.4,
  },
  {
    id: '3',
    chain1: 'DEU',
    chain2: 'FAE',
    rate: 0.3,
    amount: 17.6,
  },
  {
    id: '4',
    chain1: 'RGB',
    chain2: 'FAE',
    rate: 0.1,
    amount: 11.4,
  }
]