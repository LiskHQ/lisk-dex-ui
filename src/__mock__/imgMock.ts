export const svgUrl: string = 'svg-url';