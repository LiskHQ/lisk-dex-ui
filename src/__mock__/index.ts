export * from './governance.mock';
export * from './user.mock';