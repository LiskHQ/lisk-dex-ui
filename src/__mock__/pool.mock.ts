import { IPool } from "models";
import { mockTokens } from "./swap.mock";

export const mockPool: IPool = {
  token1: mockTokens[0],
  token2: mockTokens[1],
  token1Amount: 1862,
  token2Amount: 0.045,
  share: 0.085,
}

export const mockPools: IPool[] = [
  mockPool,
]