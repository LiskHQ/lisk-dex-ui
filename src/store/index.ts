import { combineReducers, configureStore } from '@reduxjs/toolkit';
import createSagaMiddleware from 'redux-saga';
import {
  transactionActions,
  transactionReducer,
  proposalActions,
  proposalReducer,
  poolReducer,
  poolActions,
  walletReducer,
  walletActions,
  tokenReducer,
  tokenActions
} from './slices';
import rootSaga from './sagas';

const sagaMiddleware = createSagaMiddleware();

const reducer = combineReducers({
  transaction: transactionReducer,
  proposal: proposalReducer,
  pool: poolReducer,
  wallet: walletReducer,
  token: tokenReducer,
});

export const store = configureStore({
  reducer,
  middleware: (getDefaultMiddleware) =>
    getDefaultMiddleware({ serializableCheck: false, thunk: false })
      .concat(sagaMiddleware)
});

export const AppActions = {
  transaction: transactionActions,
  proposal: proposalActions,
  pool: poolActions,
  wallet: walletActions,
  token: tokenActions,
};

sagaMiddleware.run(rootSaga);

// Infer the `RootState` and `AppDispatch` types from the store itself
export type RootState = ReturnType<typeof store.getState>;
// Inferred type: {posts: PostsState, comments: CommentsState, users: UsersState}
export type AppDispatch = typeof store.dispatch;
