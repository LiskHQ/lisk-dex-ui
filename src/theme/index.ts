export { default as theme } from "./theme";
export { default as variables } from "./variables";
