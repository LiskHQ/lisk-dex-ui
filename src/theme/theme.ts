import {
  createTheme,
  responsiveFontSizes,
  ThemeOptions,
} from "@mui/material/styles";

import variables from "./variables";

const options: ThemeOptions = {
  palette: {
    primary: variables.primary,
    secondary: variables.secondary,
  },
  typography: {
    h1: {
      fontSize: "2.5rem",
      fontWeight: 600,
    },
    h2: {
      fontSize: "2rem",
      fontWeight: 400,
    },
    h3: {
      fontSize: "1.75rem",
      fontWeight: 400,
    },
    h4: {
      fontSize: "1.25rem",
      fontWeight: 400,
    },
    h5: {
      fontSize: "1rem",
      fontWeight: 400,
    },
    h6: {
      fontSize: "0.75rem",
      fontWeight: 400,
    },
  },
  breakpoints: {
    values: {
      xs: 0,
      sm: 576,
      md: 768,
      lg: 1024,
      xl: 1440,
      xxl: 1950,
      xxxl: 2300,
    },
  },
};

const theme = responsiveFontSizes(createTheme(options));
export default theme;
