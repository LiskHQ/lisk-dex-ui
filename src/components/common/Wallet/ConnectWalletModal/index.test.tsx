import { ThemeProvider } from '@mui/material';
import { fireEvent, render } from '@testing-library/react';
import React from 'react';
import { lightTheme } from 'styles/theme';
import { ConnectWalletModal, IConnectWalletModalProps } from '.';
import { mockChainData, mockChainOptions } from '__mock__';
import { Provider } from 'react-redux';
import { store } from 'store';

function renderComponent(props: IConnectWalletModalProps) {
  return render(
    <Provider store={store}>
      <ThemeProvider theme={lightTheme}>
        <ConnectWalletModal {...props} />
      </ThemeProvider>
    </Provider>
  );
}

describe('ConnectWalletModal component', () => {
  const mockProps: IConnectWalletModalProps = {
    chainData: mockChainData,
    uri: '',
    chainOptions: mockChainOptions,
    connected: false,
    onConnect: jest.fn(),
    onClose: jest.fn(),
  };

  it('checks if the component matches the snapshot', () => {
    renderComponent(mockProps);
    expect(screen).toMatchSnapshot();
  });

  it('test connected', () => {
    const { getByText } = renderComponent({
      ...mockProps,
      uri: 'uri',
      connected: true,
    });
    expect(getByText('Connected')).toBeInTheDocument();
  });

  it('test connect', () => {
    const { getByTestId } = renderComponent(mockProps);
    fireEvent.click(getByTestId('button-connect-test'));
    expect(mockProps.onConnect).toBeCalled();
  });

  it('test close', () => {
    const { getByTestId } = renderComponent(mockProps);
    fireEvent.click(getByTestId('button-close-test'));
    expect(mockProps.onClose).toBeCalled();
  });
});