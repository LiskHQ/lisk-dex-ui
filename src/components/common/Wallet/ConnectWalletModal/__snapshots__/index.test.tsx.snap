// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConnectWalletModal component checks if the component matches the snapshot 1`] = `Screen {}`;
